src/cpp/server/server_cc.cc

```diff
@@ -318,8 +321,8 @@ class Server::SyncRequestThreadManager : public ThreadManager {
   }
 
   void Shutdown() override {
-    server_cq_->Shutdown();
     ThreadManager::Shutdown();
+    server_cq_->Shutdown();
```

Comment:
Just to save us a bit more time :)
In server code, it is checking ThreadManager::IsShutdown() first and then request a call on the server_cq_.